fetch-in-react-practice/client/src/components/Todos.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+const url = (path) => `${process.env.REACT_APP_BASE_URL}${path}`;
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  async function fetchData() {
+    const response = await fetch('/api/todos', { method: 'POST' });
+    if (!response.ok) throw new Error('Status', response.status);
+    const data = response.json();
+  }
+
+  function handleError(err) {
+    setError(err);
+    console.error('An error occurred during fetch.');
+  }
+
+  useEffect(() => {
+    /* Use fetch to send a GET request to `/api/todos`.
+     * Once the response JSON is received and parsed,
+     * update state with the received todos.
+     * Handle errors correctly and set `error` to any error or error status code.
+     * Set `isLoading` to false when all network activity is finished.
+     */
+    async function fetchTodos() {
+      try{
+        const response = await fetch(url('/api/todos'));
+        if (!response.ok) throw new Error('Status:', response.status);
+        const data = await response.json();
+        setTodos(data);
+      } catch(err) {
+        handleError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchTodos();
+  }, [todos]);
+
+  function addTodo(newTodo) {
+    /* Use fetch to send a POST request to `/api/todos`.
+     * Once the response JSON is received and parsed,
+     *   - set the Todos to a new array with the added Todo concatenated
+     *   - handle errors appropriate (as mentioned above)
+     *
+     * DO NOT MUTATE the original state array, nor any objects within it.
+     *
+     * TIP: When calling fetch, be sure to SERIALIZE the todo object in the body
+     * with JSON.stringify() and specify the "Content-Type" header as "application/json"
+     * TIP: The "(prev) =>" version of the state setter function might be convenient
+     * TIP: Use Array.prototype.concat to create a new array containing the contents
+     * of the old array, plus the object returned by the server.
+     */
+    async function fetchTodo() {
+      try{
+        const response = await fetch(url('/api/todos'), {
+           method: "POST", headers: {"Content-Type": "application/json" }, body: JSON.stringify(newTodo)});
+        if (!response.ok) throw new Error('Status', response.status);
+        const data = await response.json();
+        setTodos(Array.prototype.concat(todos, data));
+      } catch(err) {
+        handleError(err);
+      }
+    }
+    fetchTodo();
```

Comment:
There's no reason to do this embedded function thing on event handlers, only on `useEffect`. Event handlers can be async.